packages/sdk/clients/createNexusClient.test.ts

```diff
@@ -0,0 +1,275 @@
+      transport: http(chain.rpcUrls.default.http[0])
+    })
+
+    expect(await ethOwnerNexusClient.account.getAddress()).toBeTruthy()
+  })
+
+  test("should read estimated user op gas values", async () => {
+    const userOp = await nexusClient.prepareUserOperation({
+      calls: [
+        {
+          to: recipientAccount.address,
+          data: "0x"
+        }
+      ]
+    })
+
+    const estimatedGas = await nexusClient.estimateUserOperationGas(userOp)
+    expect(estimatedGas.verificationGasLimit).toBeTruthy()
+    expect(estimatedGas.callGasLimit).toBeTruthy()
+    expect(estimatedGas.preVerificationGas).toBeTruthy()
+  }, 30000)
+
+  test.skip("should create a smart account with paymaster with an api key", async () => {
+    const paymaster = nexusClient.paymaster
+    expect(paymaster).not.toBeNull()
+    expect(paymaster).not.toBeUndefined()
+  })
+
+  test("should return chain object for chain id 1", async () => {
+    const chainId = 1
+    const chain = getChain(chainId)
+    expect(chain.id).toBe(chainId)
+  })
+
+  test("should have correct fields", async () => {
+    const chainId = 1
+    const chain = getChain(chainId)
+    ;[
+      "blockExplorers",
+      "contracts",
+      "fees",
+      "formatters",
+      "id",
+      "name",
+      "nativeCurrency",
+      "rpcUrls",
+      "serializers"
+    ].every((field) => {
+      expect(chain).toHaveProperty(field)
+    })
+  })
+
+  test("should throw an error, chain id not found", async () => {
+    const chainId = 0
+    expect(() => getChain(chainId)).toThrow(ERROR_MESSAGES.CHAIN_NOT_FOUND)
+  })
+
+  test("should have attached erc757 actions", async () => {
+    const [
+      accountId,
+      isModuleInstalled,
+      supportsExecutionMode,
+      supportsModule
+    ] = await Promise.all([
+      nexusClient.accountId(),
+      nexusClient.isModuleInstalled({
+        module: {
+          type: "validator",
+          address: addresses.K1Validator,
+          context: "0x"
+        }
+      }),
+      nexusClient.supportsExecutionMode({
+        type: "delegatecall"
+      }),
+      nexusClient.supportsModule({
+        type: "validator"
+      })
+    ])
+    expect(accountId).toBe("biconomy.nexus.1.0.0-beta")
+    expect(isModuleInstalled).toBe(true)
+    expect(supportsExecutionMode).toBe(true)
+    expect(supportsModule).toBe(true)
+  })
+
+  test("should send eth twice", async () => {
+    const balanceBefore = await getBalance(testClient, recipientAddress)
+    const tx = { to: recipientAddress, value: 1n }
+    const hash = await nexusClient.sendTransaction({ calls: [tx, tx] })
+    const { status } = await testClient.waitForTransactionReceipt({ hash })
+    const balanceAfter = await getBalance(testClient, recipientAddress)
+    expect(status).toBe("success")
+    expect(balanceAfter - balanceBefore).toBe(2n)
+  })
+
+  // Not working
+  test.skip("should uninstall modules", async () => {
+    const result = await nexusClient.uninstallModules({
```

Comment:
this will batch uninstallModule calls?

and Module object passed here requires different info depending on type of a module. 

install requires install data. in case of fallback we need to give selector, is it a context?

in order to check if fallback is installed we pass a selector but for other we dont

what is data vs context defined here

Reply:
context = deInitData here 
@livingrockrises  I think uninstallModule and installModule also work with fallback module, you just build a context with module selector and data  and pass it as a param, it would be better to check for module type inside these methods and have a strongly typed arguments just to let the user know how to handle it, like it was previously done

Reply:
The install / unistall code hasn't been tested (again, these are not currently working prior to this refactor too). I couldn't actually write tests for this because at the time there wasn't a module that I could had access to - to install and uninstall. I said to gabi that I would revisit this once ownables have been integrated, so we have a module for testing